lib/tessel/deploy.js

```diff
@@ -84,28 +84,38 @@ function untarStdin(t, filepath, opts, callback) {
 
     var tesselClassic = require('tessel');
 
-    // Gather details about the file structure of the script being sent
-    var ret = tesselClassic.analyzeScript(process.cwd() + "/" + opts.entryPoint, {verbose: opts.verbose});
-    // Tar up the code to improve transfer rates
-    tesselClassic.tarCode(ret.pushdir, {node: true}, function(err, bundle) {
-      logs.info('Bundled. Writing to T2....', bundle.length);
-      // Throw any unfortunate errors
-      if (err) throw err;
+    // TODO: This section unnecessarily synchronous
+    t.getName(function (err, tesselname) {
+      // Gather details about the file structure of the script being sent
+      var ret = tesselClassic.analyzeScript(process.cwd() + "/" + opts.entryPoint, {verbose: opts.verbose});
+      // Tar up the code to improve transfer rates
+      tesselClassic.tarCode(ret.pushdir, {node: true}, function(err, bundle) {
+        // RAM or Flash for log
+        var memtype;
+        if (opts._[0] == 'push') {
+          memtype = 'Flash';
+        } else {
+          memtype = 'RAM';
+        }
+        // Log write
+        logs.info('Writing ' + opts.entryPoint + ' to ' + memtype + ' on ' + tesselname + ' (' + bundle.length/1000 + ' kB)....');
```

Comment:
It would be a little cleaner to separate substrings with commas instead of manually adding all of the spaces:
```.js
logs.info('Writing', opts.entryPoint, 'to' ...);
```